sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,237 @@
+  public boolean onSingleTapUp(final @Nullable MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onSingleTapUp");
+    if (decorView == null || motionEvent == null) {
+      return false;
+    }
+
+    @SuppressWarnings("Convert2MethodRef")
+    final @Nullable View target =
+      ViewUtils.findTarget(decorView, motionEvent.getX(), motionEvent.getY(),
+        view -> ViewUtils.isViewTappable(view));
+
+    if (target == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.DEBUG, "Unable to find click target. No breadcrumb captured.");
+      return false;
+    }
+
+    addBreadcrumb(target, "click");
+    return false;
+  }
+
+  @Override
+  public boolean onScroll(
+    final @Nullable MotionEvent firstEvent,
+    final @Nullable MotionEvent currentEvent,
+    final float distX,
+    final float distY
+  ) {
+    final View decorView = ensureWindowDecorView("onScroll");
+    if (decorView == null || firstEvent == null) {
+      return false;
+    }
+
+    if (scrollState.type == null) {
+      @SuppressWarnings("Convert2MethodRef")
+      final @Nullable View target =
+        ViewUtils.findTarget(decorView, firstEvent.getX(), firstEvent.getY(),
+          view -> ViewUtils.isViewScrollable(view));
+
+      if (target == null) {
+        options
+          .getLogger()
+          .log(SentryLevel.DEBUG, "Unable to find scroll target. No breadcrumb captured.");
+        return false;
+      }
+
+      scrollState.setTarget(target);
+      scrollState.type = "scroll";
+    }
+    return false;
+  }
+
+  @Override
+  public boolean onFling(
+    final @Nullable MotionEvent motionEvent,
+    final @Nullable MotionEvent motionEvent1,
+    final float v,
+    final float v1
+  ) {
+    scrollState.type = "swipe";
+    return false;
+  }
+
+  @Override
+  public void onShowPress(MotionEvent motionEvent) {
+  }
+
+  @Override
+  public void onLongPress(MotionEvent motionEvent) {
+  }
+
+  // region utils
+  private void addBreadcrumb(
+    final @NotNull View target,
+    final @NotNull String eventType
+  ) {
+    addBreadcrumb(target, eventType, Collections.emptyMap());
+  }
+
+  private void addBreadcrumb(
+    final @NotNull View target,
+    final @NotNull String eventType,
+    final @NotNull Map<String, Object> additionalData
+  ) {
+    @NotNull String className;
+    if (target.getClass().getCanonicalName() != null) {
+      className = target.getClass().getCanonicalName();
+    } else {
+      className = target.getClass().getSimpleName();
+    }
+
+    final Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setType("user");
+    breadcrumb.setCategory("ui." + eventType);
+    breadcrumb.setData("view.id", ViewUtils.getResourceId(target));
+    breadcrumb.setData("view.class", className);
+    for (final Map.Entry<String, Object> entry : additionalData.entrySet()) {
+      breadcrumb.getData().put(entry.getKey(), entry.getValue());
+    }
+    breadcrumb.setLevel(SentryLevel.INFO);
```

Comment:
Maybe we create a `userInteraction` static method to create this breadcrumb that could be used by custom instrumentation too?